File: androidApp/src/main/java/com/app/pustakam/android/screen/bookUIView/AudioBlockView.kt
package com.app.pustakam.android.screen.bookUIView

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.app.pustakam.android.widgets.audio.AudioPlayerUIState
import com.app.pustakam.core.filesys.reader.ReaderBlock

// 📖 01-Aug-2026: AudioPlayView is used AS IS — no compact variant. Several audio blocks share a
//   page purely because their estimated heights fit, never because the widget was shrunk.
@Composable
fun AudioBlockView(block: ReaderBlock.Audio, modifier: Modifier = Modifier) {
    Column(modifier.fillMaxWidth()) {
        AudioPlayerUIState(block.item)
    }
}

File: iosApp/iosApp/view/screens/bookReader/blocks/AudioBlockView.swift
import SwiftUI
import shared

// 📖 01-Aug-2026: AudioPlayView is used AS IS — no compact variant. Several audio blocks share a
//   page purely because their estimated heights fit, never because the widget was shrunk.
struct AudioBlockView: View {
    let block: ReaderBlock.Audio
    private let mediaManager = MediaManager.mediaManager

    var body: some View {
        AudioPlayView(mediaContent: block.item)
            .frame(maxWidth: .infinity)
            .onAppear {
                if mediaManager.currentPlaying?.id != block.item.id {
                    mediaManager.prepareMedia(media: block.item)
                }
            }
    }
}
